mitiq/calibration/calibration.py

```diff
@@ -0,0 +1,204 @@
+            Executor(ideal_executor)
+            if ideal_executor and not isinstance(ideal_executor, Executor)
+            else ideal_executor
+        )
+        self.settings = settings
+        self.circuits = self.settings.make_circuits()
+        self.results: list[dict[str, Any]] = []
+
+    def get_cost(self) -> dict[str, int]:
+        """Returns the expected number of noisy expectation values required
+        for calibration. If an ideal_executor was used in specifying the
+        Calibrator object, the number of classical simulations is also
+        returned."""
+        num_circuits = len(self.circuits)
+        num_methods = len(self.settings.techniques)
+        num_options = prod(map(len, self.settings.technique_params.values()))
+
+        noisy = num_circuits * num_methods * num_options
+        ideal = noisy if self.ideal_executor else 0
+        return {
+            "noisy_executions": noisy,
+            "ideal_executions": ideal,
+        }
+
+    def run_circuits(self) -> list[dict[str, Any]]:
+        """Run the calibration circuits and store the ideal and noisy
+        expectation values for each circuit in `self.results`."""
+        expvals = []
+        for circuit_data in self.circuits:
+            circuit = circuit_data.circuit
+            circuit.append(cirq.measure(circuit.all_qubits()))
+
+            noisy_results = cast(
+                MeasurementResult, self.executor._run([circuit])[0]
+            ).result
+            noisy_distribution = bitstrings_to_distribution(noisy_results)
+            ideal_results = (
+                cast(
+                    MeasurementResult, self.ideal_executor._run([circuit])[0]
+                ).result
+                if self.ideal_executor
+                else None
+            )
+            ideal_distribution = (
+                bitstrings_to_distribution(ideal_results)
+                if ideal_results
+                else {"1" * len(circuit.all_qubits()): 1.0}
+            )
+
+            mitigated: dict[str, dict[str, Any]] = {
+                technique: {"results": [], "method_improvement_factor": None}
+                for technique in self.settings.techniques
+            }
+            for strategy in self.settings.make_strategies():
+                most_probable_bitstring = max(
+                    ideal_distribution, key=ideal_distribution.get
+                )
+                expval_executor = bitstring_executor_to_expval_executor(
+                    self.executor, most_probable_bitstring
+                )
+                mitigated_expval = strategy.mitigation_function(
+                    circuit, expval_executor
+                )
+                mitigated[strategy.technique]["results"].append(
+                    {
+                        "circuit_type": circuit_data.type,
+                        "mitigated_value": mitigated_expval,
+                        **strategy.technique_params,
+                    }
```

Comment:
For each result we are saving the circuit type but additional circuit information is lost (e.g. number of qubits, depth, or the circuit itself). This may be good or bad depending on a trade off between completeness and efficiency. 

In the future we could save here the full `circuit_data` object or, perhaps better, we could save the results `mitigated` inside the existing `circuit_data` object.

This is just a comment to think about for the future. It's probably ok to keep the current structure for now.